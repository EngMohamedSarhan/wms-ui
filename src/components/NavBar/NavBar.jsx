import { AppBar, Typography } from "@mui/material";
import Box from "@mui/material/Box";
import Button from "@mui/material/Button";
import React from "react";
import { Link } from "react-router-dom";

import logo from "../../assets/images/logo.png";
import { appBarHeight } from "../../styles/styles";

const NavBar = () => (
  <Box sx={{ flexGrow: 1, height: appBarHeight }}>
    <AppBar
      position="static"
      color="transparent"
      sx={{
        boxShadow: "none",
        display: "flex",
        flexDirection: "row",
        alignItems: "center",
        pl: 5,
        pr: 5,
        pt: 1,
        pb: 1,
      }}
    >
      <Link to="/">
        <img src={logo} style={{ width: 50, height: 50 }} />
      </Link>
      <Link to="/">
        <Typography
          variant="h2"
          color="primary"
          sx={(theme) => ({
            ml: 1,
            mb: 1,
            [theme.breakpoints.down("sm")]: {
              fontSize: "18px",
            },
          })}
        >
          Warehouse Management
        </Typography>
      </Link>
      <Button
        color="primary"
        sx={{ marginLeft: "auto", textTransform: "none", fontWeight: "600" }}
      >
        Sign Out
      </Button>
    </AppBar>
  </Box>
);

export default NavBar;
