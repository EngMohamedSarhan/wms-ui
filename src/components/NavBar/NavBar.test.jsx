/* eslint-disable no-undef */
import { fireEvent, render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import React from "react";
import { Router } from "react-router-dom";
import ShallowRenderer from "react-shallow-renderer";

import NavBar from "./NavBar";

describe("NavBar Component", () =>
  test("NavBar action", () => {
    const history = createMemoryHistory({ initialEntries: ["/lpn-date-check"] });
    const { getByAltText, getByTestId } = render(
      <Router location={history.location} navigator={history}>
        <NavBar />
      </Router>
    );
    expect(history.location.pathname).toBe("/lpn-date-check");
    const logoButton = getByAltText("logo");
    fireEvent.click(logoButton);
    expect(history.location.pathname).toBe("/");
    history.push("/lpn-date-check");
    const logoTextButton = getByTestId("logo-text-link");
    fireEvent.click(logoTextButton);
    expect(history.location.pathname).toBe("/");
  }));

describe("NavBar Component", () =>
  test("NavBar snapshot", () => {
    const renderer = new ShallowRenderer();
    const history = createMemoryHistory({ initialEntries: ["/lpn-date-check"] });
    renderer.render(
      <Router location={{ ...history.location, key: "1" }}>
        <NavBar />
      </Router>
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  }));
