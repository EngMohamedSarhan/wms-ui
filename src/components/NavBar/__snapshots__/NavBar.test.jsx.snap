// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component NavBar snapshot 1`] = `
<Context.Provider
  value={
    Object {
      "basename": "/",
      "navigator": undefined,
      "static": false,
    }
  }
>
  <Context.Provider
    value={
      Object {
        "location": Object {
          "hash": "",
          "key": "1",
          "pathname": "/lpn-date-check",
          "search": "",
          "state": null,
        },
        "navigationType": "POP",
      }
    }
  >
    <NavBar />
  </Context.Provider>
</Context.Provider>
`;

exports[`NavBar snapshot NavBar snapshot 1`] = `
<Context.Provider
  value={
    Object {
      "basename": "/",
      "navigator": undefined,
      "static": false,
    }
  }
>
  <Context.Provider
    value={
      Object {
        "location": Object {
          "hash": "",
          "key": "1",
          "pathname": "/lpn-date-check",
          "search": "",
          "state": null,
        },
        "navigationType": "POP",
      }
    }
  >
    <NavBar />
  </Context.Provider>
</Context.Provider>
`;
